astropy/utils/masked/core.py

```diff
@@ -1113,7 +1117,13 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
             axis=axis, dtype=dtype, out=out, keepdims=keepdims, where=where
         )
         n = np.add.reduce(where, axis=axis, keepdims=keepdims)
-        result /= n
+        fully_masked_axes = n == 0
```

Comment:
I can see the case for avoiding the warning, but I quite like that the value underneath the mask is now `np.nan`. So perhaps just filtering the warning out with `np.errstate` is better. Would also need to check the impact on speed.

Reply:
There's a separate PR for discussing my idea above in https://github.com/astropy/astropy/pull/14341. Happy to run speedtests for you @mhvk! Marking this convo as resolved for this PR though.